desk/app/activity.hoon

```diff
@@ -278,6 +278,22 @@
   %+  welp
   /(scot %p our.bowl)/[dude]/(scot %da now.bowl)
   path
+++  get-vessel
+  |=  [=flag:g =ship]
+  ^-  (unit vessel:fleet:g)
+  =/  groups-running
+    =>  [scry-path=scry-path ..zuse]  ~+
+    .^(? %gu (scry-path %groups /$))
+  ?.  groups-running  ~
+  =/  group-exists
+    =>  [scry-path=scry-path flag=flag ..zuse]  ~+
+    .^(? %gx (scry-path %groups /exists/(scot %p p.flag)/[q.flag]/noun))
+  ?.  group-exists  ~
+  =/  =path
+    %+  scry-path  %groups
+    /groups/(scot %p p.flag)/[q.flag]/fleet/(scot %p ship)/vessel/noun
+  =>  [vessel=vessel:fleet:g path=path ..zuse]  ~+
+  (some .^(vessel %gx path))
```

Comment:
`scry-path` is gonna have all the core/arm's context inside of it, so you'll miss the cache if you even just change state here before calling it again.

And, for any cache hit, the base path's gonna be the same, so you can just key off that and do a single cache entry:

```suggestion
  ^-  (unit vessel:fleet:g)
  =/  base-path
    (scry-path %groups /)
  =>  [flag=flag ship=ship base-path=base-path ..zuse]  ~+
  =/  groups-running
    .^(? %gu (weld base-path /$))
  ?.  groups-running  ~
  =/  group-exists
    .^(? %gx (weld base-path /exists/(scot %p p.flag)/[q.flag]/noun))
  ?.  group-exists  ~
  %-  some
  .^  vessel  %gx
    %+  weld  base-path
    /groups/(scot %p p.flag)/[q.flag]/fleet/(scot %p ship)/vessel/noun
  ==
```